src/Microsoft.OData.Edm/ExtensionMethods/ExtensionMethods.cs

```diff
@@ -1545,6 +1545,39 @@ public static IEnumerable<IEdmOperationImport> OperationImports(this IEdmEntityC
             return container.AllElements().OfType<IEdmOperationImport>();
         }
 
+        /// <summary>
+        /// Returns NavigationPropertyBindings with their corresponding container elements (EntitySet or Singelton)
+        /// </summary>
+        /// <param name="container">Reference to the calling object.</param>
+        /// <returns>collection of pairs of container element and NavigationPropertyBindings.</returns>
+        public static IEnumerable<(IEdmEntityContainerElement, IEdmNavigationPropertyBinding)> GetNavigationPropertyBindings(this IEdmEntityContainer container)

```

Comment:
Will the caller be interested in the specific kind of `IEdmEntityContainerElement` returned for each element? We are already casting inside this method, it would be a shame for the caller to have to do so as well if we can avoid it

Reply:
it is true that a user would need to cast the first element of the pair if they are interested in more than what IEdmEntityContainerElement provides (which is the name, container, kind and a few extension methods).
But I doubt that this is common since this API is specifically about all bindings, independent of the type of "bindable" container element.